domain/application/charm/types.go

```diff
@@ -32,6 +35,17 @@ const (
 	CharmHubSource CharmSource = "charmhub"
 )
 
+func FromLegacySchema(s string) CharmSource {
+	switch s {
+	case "local":
+		return LocalSource
+	case "ch":
+		return CharmHubSource
+	default:
+		return CharmHubSource
+	}
+}
```

Comment:
```suggestion
func ParseCharmSchema(s string) (CharmSource, error) {
	switch s {
	case "local":
		return LocalSource, nil
	case "ch":
		return CharmHubSource, nil
	default:
		return "", fmt.Errorf("unknown charm source %q", s)
	}
}
```

Add a comment as well!

Reply:
I specifically created this method to be used until we finish double writing, and it's explicitly called "From legacy" because it'll only map the legacy `ch` and not parse the current one `charmhub`. I agree that we should have a "Parse..." method later. 
What do you think

Reply:
I can see this being really useful, let's make it proper

Reply:
Sure! see 8c9286fa4a